shimmingtoolbox/optimizer/basic_optimizer.py

```diff
@@ -109,15 +132,48 @@ def merge_coils(self, unshimmed, affine):
 
             # Resample a coil on the unshimmed image
             resampled_coil = resample_from_to(nii_coil, nii_unshimmed).get_fdata()
+            coil_profiles_list.append(resampled_coil)
 
+        coil_profiles = np.concatenate(coil_profiles_list, axis=3)
+
+        bounds = self.merge_bounds()
+
+        return coil_profiles, bounds
+
+    def merge_bounds(self):
+        """
+        Merge the coil profile bounds into a single array.
+
+        Returns:
+            list: list of bounds corresponding to each merged coils
+        """
+
+        bounds = []
+        for coil in self.coils:
             # Concat coils and bounds
-            coil_profiles_list.append(resampled_coil)
             for a_bound in coil.coef_channel_minmax:
                 bounds.append(a_bound)
 
-        coil_profiles = np.concatenate(coil_profiles_list, axis=3)
+        return bounds
 
-        return coil_profiles, bounds
+    def initial_guess_half_bounds(self):
+        """
+        Calculates the initial guess from the bounds, sets it to half of the bounds
```

Comment:
It does not seem to set it to half the bounds but to the average between the bounds